homeassistant/components/directv/__init__.py

```diff
@@ -92,3 +84,36 @@ def get_dtv_data(
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+class DIRECTVEntity(Entity):
+    """Defines a base DirecTV entity."""
+
+    def __init__(self, *, dtv: DIRECTV, name: str, address: str = "0") -> None:
+        """Initialize the DirecTV entity."""
+        self._address = address
+        self._device_id = address if address != "0" else dtv.device.info.receiver_id
+        self._is_client = address != "0"
+        self._name = name
+        self.dtv = dtv
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return self._name
+
+
+class DIRECTVDeviceEntity(DIRECTVEntity):
```

Comment:
Why is this not part of the main entity ?

Reply:
I guess it could be since since any entity should be tied to device. Mostly following pattern of other integrations